### PR TITLE
Share the PrivateKeyInfo structure among RSA and ECC implementation

### LINKED ISSUE #10: Use generic PrivateKeyInfo
Currently, there is separate copy of PrivateKeyInfo in RSA and ECC code, handling the wrapping and unwrapping of private keys.

Simo suggested that this would be something that could be used with `#defined_by(OID)` of rust-asn1 library to provide generic structure. The WIP branch implementing this is available here (but still has some memory ownership issues to resolve):

https://github.com/Jakuje/kryoptic/commits/defined-by/

              sounds like this is a place where to use:
#[defined_by(private_key_algorithm)]

See: https://docs.rs/asn1/latest/asn1/#defined-by

_Originally posted by @simo5 in https://github.com/latchset/kryoptic/pull/5#discussion_r1507927653_

### PR DESCRIPTION
**Pull request by Jakuje**
~Using the `defined-by` magic.~

Using opaque OCTET STRING

Fixes #10.

### CODE REVIEW

**Comment by Jakuje** [comment]
Rewritten to use opaque OCTET STRING in the shared code as the DEFINED by would need us to share all algorithm structures. Should be ready.

File: src/kasn1.rs [hunk 1]
```diff
@@ -123,3 +123,44 @@ impl<'a> asn1::SimpleAsn1Writable for DerEncOctetString<'a> {
...
+    version: Version,
+    private_key_algorithm: asn1::ObjectIdentifier,
+    private_key: DerEncOctetString<'a>,
+    #[explicit(1)]
+    attributes: Option<Attributes<'a>>,
+}
+
+impl PrivateKeyInfo<'_> {
+    pub fn new<'a>(
+        private_key_asn1: &'a Vec<u8>,
```
Comment (simo5) [inline src/kasn1.rs:148]: Why not pass directly a slice here?

**Review by simo5 — APPROVED** [review]
Just one nit but otherwise this looks good and simple enough!

File: src/kasn1.rs (hunk 1 above)
Comment (Jakuje) [reply src/kasn1.rs]: I think I got confused with the lifetimes. Changed to use the `&[u8]`. Its indeed simpler.

**Review by simo5 — APPROVED** [review]
LGTM

**Pull request merged**